linux/keyman-config/keyman_config/__init__.py

```diff
@@ -44,15 +46,22 @@ def _(txt):
     try:
         # Try new sentry-sdk first
         sentry_sdk = importlib.import_module('sentry_sdk')
-        from sentry_sdk import configure_scope
+        from sentry_sdk import configure_scope, set_user
+        from sentry_sdk.integrations.logging import LoggingIntegration
         HaveSentryNewSdk = True
 
+        sentry_logging = LoggingIntegration(
+            level=logging.INFO,          # Capture info and above as breadcrumbs
+            event_level=logging.CRITICAL # Send critical errors as events
+        )
         SentryUrl = "https://1d0edbf2d0dc411b87119b6e92e2c357@sentry.keyman.com/12"
         sentry_sdk.init(
             dsn=SentryUrl,
             environment=__environment__,
             release=__version__,
+            integrations=[sentry_logging],
         )
+        set_user({'username': getpass.getuser()})
```

Comment:
I'm a bit torn about logging the username in plain text. In general I think it's helpful because it makes it easier to see in the Sentry UI how many users are affected, and having the username in plain text might help in identifying and helping a particular user. We could also just log a hash value of the username which would improve privacy, but wouldn't allow to connect an event with a particular user.

Reply:
fwiw, the Sentry projects keyman-android separates events by `user.id` while keyman-web uses `user.ip`.

Reply:
>  We could also just log a hash value of the username which would improve privacy, but wouldn't allow to connect an event with a particular user.

Let's do that. If we are wanting to link to a specific user, that information should come from them via other channels rather than being in the automatic report.  For example, they may send us a crash id and we can annotate the crash report with their details at that time.

Reply:
:heavy_check_mark: